src/test/rpc/AccountObjects_test.cpp

```diff
@@ -1032,13 +1032,152 @@ class AccountObjects_test : public beast::unit_test::suite
         BEAST_EXPECT(acct_objs_is_size(acct_objs(gw, jss::hashes), 0));
     }
 
+    void
+    testNFTsMarker()
+    {
+        // there's some bug found in account_nfts method that it did not
+        // return invalid params when providing unassociated nft marker.
+        // this test tests both situations when providing valid nft marker
+        // and unassociated nft marker.
+        testcase("NFTsMarker");
+
+        using namespace jtx;
+        Env env(*this);
+
+        Account const bob{"bob"};
+        env.fund(XRP(10000), bob);
+
+        // this lambda function is used to create some fake marker using given
+        // taxon and sequence because we want to test some unassociated markers
+        // later
+        auto createFakeNFTMarker = [](AccountID const& issuer,
+                                      uint32_t taxon,
+                                      uint32_t tokenSeq,
+                                      std::uint16_t flags = 0,
+                                      std::uint16_t fee = 0) {
+            taxon = boost::endian::native_to_big(taxon);
+            tokenSeq = boost::endian::native_to_big(tokenSeq);
+            flags = boost::endian::native_to_big(flags);
+            fee = boost::endian::native_to_big(fee);
+            std::array<std::uint8_t, 32> buf{};
+            auto ptr = buf.data();
+            std::memcpy(ptr, &flags, sizeof(flags));
+            ptr += sizeof(flags);
+            std::memcpy(ptr, &fee, sizeof(fee));
+            ptr += sizeof(fee);
+            std::memcpy(ptr, issuer.data(), issuer.size());
+            ptr += issuer.size();
+            std::memcpy(ptr, &taxon, sizeof(taxon));
+            ptr += sizeof(taxon);
+            std::memcpy(ptr, &tokenSeq, sizeof(tokenSeq));
+            ptr += sizeof(tokenSeq);
+            return uint256::fromVoid(buf.data());
+        };
+
+        // this lambda function is used to check if the account_nfts method
+        // returns the correct token information. lastIndex is used to query the
+        // last marker.
+        auto compareNFTs = [&](unsigned const limit,
+                               unsigned const lastIndex,
+                               unsigned const nftsSize,
+                               const std::vector<Json::Value>& tokenIDs) {
```

Comment:
Yes. I did not pass in them in my first version. I added them because I was asked to move the lambda function to the beginning of the test case where these values are not defined in the first round review.